doc/kernel/usermode/mpu_stack_objects.rst

```diff
@@ -0,0 +1,49 @@
+.. _mpu_stack_objects:
+
+MPU Stack Objects
+#################
+
+Thread Stack Creation
+=====================
+
+Thread stacks are declared staticly with :c:macro:`K_THREAD_STACK_DEFINE()`
+or embedded within structures using c:macro:`K_THREAD_STACK_MEMBER()`
+
+For architectures which utilize memory protection unit (MPU) hardware,
+stacks are physically contiguous allocations.  This contiguous allocation
+has implications for the placement of stacks in memory, as well as the
+implementation of other features such as stack protection and userspace.
+
+Stack Guards
+============
+
+Stack protection mechanisms require hardware support that can restrict access
+to memory.  Memory protection units can provide this kind of support.
+The MPU provides a fixed number of regions.  Each region contains information
+about the start, end, size, and access attributes to be enforced on that
+particuliar region.
+
+Stack guards are implemented by using a single MPU region and setting the
+attributes for that region to not allow write access.  If invalid accesses
+occur, a fault ensues.  The stack guard is defined at the bottom of the stack,
+or the lowest address.
+
+Memory Placement
+================
+
+During stack creation, a set of constraints are enforced on the allocation of
+memory.  These contraints include determining the alignment of the stack and
+the correct sizing of the stack.  During linking of the binary, these
+constraints are used to place the stacks properly.
+
+The main source of the memory constraints is the MPU design for the SoC.
+Some MPUs require that each region be aligned to a power of two.  These SoCs
+will have :option:`CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT` defined.
+This means that a 1500 byte stack should be aligned to a 2kB boundary.  This
```

Comment:
I'd rewrite this sentence a bit, and move it to the end of this paragraph:

    This requires a power of two ceiling be computed for the alignment and size
    of the stack, and can result is the actual memory allocation of a given stack
    being larger than the requested size.  For example, a request for a 1500 byte stack
    would be aligned and resized to a 2kB boundary and size if a power of two alignment
    is required by the MPU.

Reply:
i redid the section.   Take a look at the new version.